app/services/account_membership_cloner.rb

```diff
@@ -0,0 +1,43 @@
+class AccountMembershipCloner
+
+  PROTECTED_USER_ROLE = "Owner".freeze
+  ALT_FOR_PROTECTED_ROLE = "Business Administrator".freeze
+
+  attr_reader :error
+
+  def initialize(account_users_to_clone:, clone_to_user:)
+    @account_users_to_clone = account_users_to_clone
+    @clone_to_user = clone_to_user
+  end
+
+  def perform
+    perform!
+  rescue => e
+    @error = e
+    false
+  end
+
+  private
+
+  def perform!
+    ActiveRecord::Base.transaction do
+      @account_users_to_clone.each do |au|
+        create_account_user(au.attributes)
+      end
+    end
+  end
+
+  def create_account_user(attributes)
+    AccountUser.create!(
```

Comment:
We need to create a `LogEvent` when we do this as well.

Reply:
I'm unsure if we need to send an email to the account owners. I feel like we will want to do that, but we'll need to think about it and if there's a way to roll up the emails.